### PR TITLE
ArticleCard

### PR DESCRIPTION
**Pull request by amoedoamorim**
## Description

Refactor `FactCheckCard` to be `ArticleCard` with `explainer` and `fact-check` variants.
`FactCheckUrl` followed suit.

References: CV2-4438

## Type of change

- [x] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Used component in Sandbox and made sure unit test is passing.
I'm working to improve unit tests with to test each variant.

## Things to pay attention to during code review

Made the `ArticleUrl` component self-manage its rendering by returning `null` when there's no url.
I find this approach cleaner than doing the conditional on the parent component.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [x] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Comment by caiosba** [comment]
Yes, I'm still reviewing the code but that would be my first question - if the changes here are good to go as a refactoring of if they should be against the EPIC branch.

**Comment by amoedoamorim** [comment]
Yeah, they should be self-contained and safe as a refactoring targeted for develop, though

File: src/app/components/search/SearchResultsCards/ArticleCard.js [hunk 1]
```diff
@@ -51,21 +70,25 @@ const FactCheckCard = ({
...
   statusLabel: PropTypes.string.isRequired,
   statusColor: PropTypes.string,
   date: PropTypes.number.isRequired, // Timestamp
   summary: PropTypes.string,
   url: PropTypes.string,
-  teamAvatar: PropTypes.string.isRequired, // URL
-  teamName: PropTypes.string.isRequired,
+  teamAvatar: PropTypes.string, // URL
+  teamName: PropTypes.string,
+  variant: PropTypes.oneOf(['explainer', 'fact-check']),
```
Comment (caiosba) [inline src/app/components/search/SearchResultsCards/ArticleCard.js:91]: There are some props missing to be declared here:

-   languageCode
-   tags
-   onChangeTags

**Comment by caiosba** [comment]
> Yeah, they should be self-contained and safe as a refactoring targeted for develop, though

In that case, I suggest we keep against `develop` - the closer the EPIC branch gets to `develop`, the better, actually.

File: src/app/components/search/SearchResultsCards/SharedItemCardFooter.js [hunk 2]
```diff
@@ -74,6 +83,7 @@ SharedItemCardFooter.propTypes = {
   mediaType: PropTypes.string,
   requestsCount: PropTypes.number,
   suggestionsCount: PropTypes.number,
+  languageCode: PropTypes.string,
```
Comment (caiosba) [inline src/app/components/search/SearchResultsCards/SharedItemCardFooter.js:86]: Should add `tags` to `propTypes` too.

**Review by caiosba — CHANGES_REQUESTED** [review]

**Comment by caiosba** [comment]
Code reviewed, trying it locally now.

**Comment by caiosba** [comment]
Works well locally! Tried on a feed sharing only fact-checks.

**Pull request merged**